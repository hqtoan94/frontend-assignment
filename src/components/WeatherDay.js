import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-bootstrap';

const weekday = new Array(7);
weekday[0] = "Sunday";
weekday[1] = "Monday";
weekday[2] = "Tuesday";
weekday[3] = "Wednesday";
weekday[4] = "Thursday";
weekday[5] = "Friday";
weekday[6] = "Saturday";

export default function WeatherDay(props) {
  const { weather } = props;
  const day = new Date(weather.applicable_date).getDay();

  return (
    <Col xs={12} sm={6} md={4} lg={2} className="text-center mt-4">
      <div className="border rounded p-3">
        <p>{weekday[day]}</p>
        <img className="w-50" alt="Weather icon" src={`https://www.metaweather.com/static/img/weather/${weather.weather_state_abbr}.svg`} />
        <p className="mt-3 mb-0">Min: {parseInt(weather.min_temp)}</p>
        <p className="mb-0">Max: {parseInt(weather.max_temp)}</p>
      </div>
    </Col>
  )
}

WeatherDay.propTypes = {
  weather: PropTypes.shape({
    weather_state_name: PropTypes.string.isRequired,
    weather_state_abbr: PropTypes.string.isRequired,
    applicable_date: PropTypes.string.isRequired,
    min_temp: PropTypes.number.isRequired,
    max_temp: PropTypes.number.isRequired
  })
};
