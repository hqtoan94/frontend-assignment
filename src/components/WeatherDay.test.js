import React from 'react';
import { render } from '@testing-library/react';
import WeatherDay from './WeatherDay';

const weather = {
  id: 4614983327416320,
  weather_state_name: "Heavy Rain",
  weather_state_abbr: "hr",
  applicable_date: "2020-05-20",
  min_temp: 27.9,
  max_temp: 33.815
};

test('renders correctly and without crashing', () => {
  const { container } = render(<WeatherDay weather={weather} />);
  expect(container).toMatchSnapshot();
});
