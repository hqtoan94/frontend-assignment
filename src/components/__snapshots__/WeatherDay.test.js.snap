// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and without crashing 1`] = `
<div>
  <div
    class="text-center mt-4 col-lg-2 col-md-4 col-sm-6 col-12"
  >
    <div
      class="border rounded p-3"
    >
      <p>
        Wednesday
      </p>
      <img
        alt="Weather icon"
        class="w-50"
        src="https://www.metaweather.com/static/img/weather/hr.svg"
      />
      <p
        class="mt-3 mb-0"
      >
        Min: 
        27
      </p>
      <p
        class="mb-0"
      >
        Max: 
        33
      </p>
    </div>
  </div>
</div>
`;
